core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.component.ts

```diff
@@ -367,9 +373,30 @@ export class TranslationSuggestionReviewModalComponent implements OnInit {
     }
   }
 
+  isImageRemoved(): boolean {
+    const imageTags = this.editedContent.html.match(this.IMAGE_TAG_REGEX);
+    const currentImageCount = imageTags ? imageTags.length : 0;
+    return currentImageCount < this.initialImageCount;
+  }
+
+  get isUpdateDisabled(): boolean {
+    return this.startedEditing && this.isImageRemoved();
+  }
+
+  public getImageTagRegex(): RegExp {
+    return this.IMAGE_TAG_REGEX;
+  }
```

Comment:
Why is this a public method if it's not being used in the component's template or outside the component? Also per Sean's [comment](https://github.com/oppia/oppia/pull/21241#discussion_r1877423146), I think it might be better to refactor the HTML parsing functionality for images/videos/etc into a separate service, so other teams can use it.

Reply:
@StephenYu2018 , thanks for pointing that out! The `getImageTagRegex` method is currently public because it's being accessed in the frontend test cases.

That said, I agree with Sean's suggestion to refactor the HTML parsing functionality into a separate HtmlParsingService. Once the refactor is complete, I will update the test cases to use the new service.

Reply:
Just a general note, you shouldn't make a function public just for the purposes of tests. Test the stuff within the private function through the public API, instead.

Reply:
Thanks @seanlip ! I'll make the changes.